src/libs/actions/IOU.ts

```diff
@@ -4573,6 +4626,15 @@ function updateMoneyRequestDate(
         data = getUpdateTrackExpenseParams(transactionID, transactionThreadReportID, transactionChanges, policy);
     } else {
         data = getUpdateMoneyRequestParams(transactionID, transactionThreadReportID, transactionChanges, policy, policyTags, policyCategories);
+        const currentTransactionViolations = allTransactionViolations[`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`];
```

Comment:
I see we have this logic in all 3 callers of `updateDuplicateTransactionViolation` (well, not 100% sure the one in line 7884 is exactly the same), can't we move the logic to `updateDuplicateTransactionViolation` instead?